core/trino-grammar/src/main/antlr4/io/trino/grammar/sql/SqlBase.g4

```diff
@@ -733,7 +734,12 @@ normalForm
 
 type
     : ROW '(' rowField (',' rowField)* ')'                                         #rowType
-    | INTERVAL from=intervalField (TO to=intervalField)?                           #intervalType
+    | INTERVAL from=YEAR (TO to=MONTH)?                                            #yearMonthIntervalDataType
+    | INTERVAL from=MONTH                                                          #yearMonthIntervalDataType
+    | INTERVAL from=DAY (TO to=(HOUR | MINUTE | SECOND))?                          #dayTimeIntervalDataType
+    | INTERVAL from=HOUR (TO to=(MINUTE | SECOND))?                                #dayTimeIntervalDataType
+    | INTERVAL from=MINUTE (TO to=MINUTE)?                                         #dayTimeIntervalDataType
+    | INTERVAL from=SECOND                                                         #dayTimeIntervalDataType
```

Comment:
why is the sign part missing here?

Reply:
Also since you've changed the type names here I guess `AstBuilder` also needs to be updated.